File: ProiectPPMG/java1/VariantaSeriala.java
/**
 * Created by Bianca on 5/5/2019.
 */
public class VariantaSeriala {
    public static void main (String [] args){

        int s = 0;
        int a = 0;
        long b = 35000000;
        int copie;
        long startTime = System.nanoTime();
        for(int i = a ; i<b; i++){
            if(isPrime1(i)){
                for(int j = 1; j<= lengthNum(i); j++)
                    for(int k = 0; k < 10; k++){
                        copie = replaceDigitFromPosition(i, j, k);
                        if(isPrime1(copie) == false){
                            s += 1;
                        }
                    }
                if(s == 9* lengthNum(i)){
                    System.out.println(i);
                }
                s = 0;
            }
        }
        long endTime = System.nanoTime();
        long timeElapsed = endTime - startTime;

        System.out.println("Execution time in nanoseconds  : " + timeElapsed);

        double elapsedTimeInSecond = (double) timeElapsed / 1_000_000_000;
        System.out.println("Execution time in seconds:" + String.format("%.4f", elapsedTimeInSecond));

    }

    private static boolean isPrime1(int num){
        if ( num > 2 && num%2 == 0 ) {
            return false;
        }
        int top = (int)Math.sqrt(num) + 1;
        for(int i = 3; i < top; i+=2){
            if(num % i == 0){

                return false;
            }
        }
        return true;
    }
    private static int lengthNum(int n){
        int len = 0;
        if ( n > 0 ){
            len = (int)(Math.log10(n)+1);
        }
        return len;
    }
    static int replaceDigitFromPosition(int x, int pos, int digit)
    {
        int copie = 0;

        if(pos == 1){
            copie += (int)(x / Math.pow(10, pos)) * 10 + digit;
        }else if(pos > 1 && pos < lengthNum(x)){
            copie += (int)(x / Math.pow(10.0, pos)) * Math.pow(10.0, pos) + digit * Math.pow(10.0, pos-1) + x % Math.pow(10.0, pos-1);
        }else
        {
            copie += digit * Math.pow(10.0, pos-1) + x % Math.pow(10.0, pos-1);
        }
        return copie;
    }
}

File: ProiectPPMG/OpenMP/ExempluOpenMp/ExempluOpenMp/VariantaSeriala.cpp
#include "stdafx.h"
#include <omp.h>
#include <stdio.h>
#include <math.h>
#include <time.h>

bool isPrime11(int num) {
	if (num > 2 && num % 2 == 0) {
		return false;
	}
	int top = (int)sqrt(num) + 1;
	for (int i = 3; i < top; i += 2) {
		if (num % i == 0) {

			return false;
		}
	}
	return true;
}

int lengthNum1(int n) {
	int c = 0;
	while (n != 0)
	{
		n /= 10;
		++c;
	}
	return c;
}
int replaceDigitFromPosition1(int x, int pos, int digit) {
	int copie = 0;

	if (pos == 1) {
		copie += (int)(x / pow(10, pos)) * 10 + digit;
	}
	else if (pos > 1.0 && pos < lengthNum1(x)) {
		copie += (int)(x / pow(10.0, pos)) * pow(10.0, pos) + digit * pow(10.0, pos - 1.0) + x % (int)pow(10.0, pos - 1);
	}
	else
	{
		copie += digit * pow(10.0, pos - 1) + x % (int)pow(10.0, pos - 1);
	}
	return copie;

}

void main1()
{
	int s = 0;
	int a = 1;
	int b = 4393139;
	int copie;
	int i, j, k;

	for (i = a; i<b; i++)
	{
		if (isPrime11(i)) {
			for (j = 1; j <= lengthNum1(i); j++)
				for (k = 0; k < 10; k++) {
					copie = replaceDigitFromPosition1(i, j, k);
					if (isPrime11(copie) == false) {
						s += 1;
					}
				}
			if (s == 9 * lengthNum1(i)) {
				printf("%d\n", i);
			}
			s = 0;
		}
	}


}